patches/tModLoader/Terraria/ModLoader/TMLLootDatabase.cs

```diff
@@ -0,0 +1,831 @@
+			shop.Add(588);
+			shop.Add(589);
+			shop.Add(590);
+			shop.Add(597);
+			shop.Add(598);
+			shop.Add(596);
+			for (int i = 1873; i < 1906; i++) {
+				shop.Add(i);
+			}
+			RegisterNpcShop(9, shop);
+		}
+
+		private void RegisterTruffle() {
+			ChestLoot shop = new();
+			shop.Add(756, ChestLoot.Condition.DownedMechBossAny);
+			shop.Add(787, ChestLoot.Condition.DownedMechBossAny);
+			shop.Add(868);
+			shop.Add(1551, ChestLoot.Condition.DownedPlantera);
+			shop.Add(1181);
+			shop.Add(783);
+			RegisterNpcShop(10, shop);
+		}
+
+		private void RegisterSteampunker() {
+			ChestLoot shop = new();
+			shop.Add(779);
+			shop.Add(748, new ChestLoot.Condition(NetworkText.Empty, () => Main.moonPhase >= 4));
+			for (int i = 0; i < 3; i++) {
+				shop.Add(839 + i, new ChestLoot.Condition(NetworkText.Empty, () => Main.moonPhase < 4));
+			}
+			shop.Add(948, ChestLoot.Condition.DownedGolem);
+			shop.Add(3623);
+			shop.Add(3603);
+			shop.Add(3604);
+			shop.Add(3607);
+			shop.Add(3605);
+			shop.Add(3606);
+			shop.Add(3608);
+			shop.Add(3618);
+			shop.Add(3602);
+			shop.Add(3663);
+			shop.Add(3609);
+			shop.Add(3610);
+			shop.Add(995);
+			shop.Add(2203, ChestLoot.Condition.DownedEyeOfCthulhu, ChestLoot.Condition.DownedEowOrBoc, ChestLoot.Condition.DownedSkeletron);
+			shop.Add(2193, ChestLoot.Condition.CrimsonWorld);
+			shop.Add(4142, ChestLoot.Condition.CorruptionWorld);
+			shop.Add(2192, ChestLoot.Condition.InGraveyard);
+			shop.Add(2204, ChestLoot.Condition.InJungleBiome);
+			shop.Add(2198, ChestLoot.Condition.InSnowBiome);
+			shop.Add(2197, new ChestLoot.Condition(NetworkText.Empty, () => (Main.LocalPlayer.position.Y / 16.0) < Main.worldSurface * 0.3499999940395355));
+			shop.Add(2196, new ChestLoot.Condition(NetworkText.Empty, () => Main.LocalPlayer.HasItem(832)));
+			shop.Add(1263);
+
+			ChestLoot.Entry entry = new(new ChestLoot.Condition(NetworkText.Empty, () => Main.eclipse || Main.bloodMoon));
+			entry.OnSuccess(784, ChestLoot.Condition.CrimsonWorld);
+			entry.OnSuccess(782, ChestLoot.Condition.CorruptionWorld);
+			entry.OnFail(781, ChestLoot.Condition.InHallowBiome);
+			entry.OnFail(780, ChestLoot.Condition.InHallowBiome);
+			shop.Add(entry);
+
+			shop.Add(1344, ChestLoot.Condition.Hardmode);
+			shop.Add(4472, ChestLoot.Condition.Hardmode);
+			shop.Add(1742, ChestLoot.Condition.Halloween);
+			RegisterNpcShop(11, shop);
+		}
+
+		private void RegisterDyeTrader() {
+			ChestLoot shop = new();
+			shop.Add(1037);
+			shop.Add(2874);
+			shop.Add(1120);
+			shop.Add(1969, new ChestLoot.Condition(NetworkText.Empty, () => Main.netMode == NetmodeID.MultiplayerClient));
+			shop.Add(3248, ChestLoot.Condition.Halloween);
+			shop.Add(1741, ChestLoot.Condition.Halloween);
+			shop.Add(2871, ChestLoot.Condition.IsMoonFull);
+			shop.Add(2872, ChestLoot.Condition.IsMoonFull);
+			shop.Add(4663, ChestLoot.Condition.BloodMoon);
+			shop.Add(4662, ChestLoot.Condition.InGraveyard);
+			RegisterNpcShop(12, shop);
+		}
```

Comment:
Can we get ExamplePerson to be updated with an example like this in Setup Shops?

Just a brief couple of Example items being added with some standard conditions, and setting the shop name as Examplium